ci/check-protobuf-against-stable.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+eval "$(dev-env/bin/dade assist)"
+
+readonly BUF_IMAGE_TMPDIR="$(mktemp -d)"
+trap 'rm -rf ${BUF_IMAGE_TMPDIR}' EXIT
+
+readonly CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+echo "The target branch is '${SYSTEM_PULLREQUEST_TARGETBRANCH}'."
+
+# For `main` and PRs targeting `main`, we simply check against the most recent stable tag.
+#
+# For release branches and PRs targeting them, we should really check against all the most recent stable tags
+# reachable from either the current branch or from previous release branches (say, against both `1.17.1` and
+# `1.16.2` created after the `release/1.17.x` branch).
+#
+# Instead, we approximate by checking only against the most recent stable tag reachable from the current branch,
+# under the assumption that if a lesser release branch contains a protobuf change, then it will also be present in
+# higher ones either through a shared commit or a back-port from `main`.
+readonly RELEASE_BRANCH_REGEX="^release/.*"
+GIT_TAG_SCOPE=""
+if [[ "${CURRENT_BRANCH}" =~ ${RELEASE_BRANCH_REGEX} || "${SYSTEM_PULLREQUEST_TARGETBRANCH}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
```

Comment:
This will not run on release branches, so I'm not sure what the point of the first condition is here. The only case in which `CURRENT_BRANCH` would match the regex is if we deliberately try to trick it by creating a PR **from** a `release/x` branch, which should never happen.

Reply:
Nothing runs on release branches because we have enabled sequential merges there, so we believe PR checks are enough.

When I say "create a PR from a release branch" here I mean one where the _source_ branch is the release branch itself. Of course PRs _to_ release branches are made from commits that have some common ancestors with the release branch, but they are made _from_ branches that contain at least one more commit than the release branch itself.

Reply:
f7ec421